api/src/main/java/org/openmrs/module/reporting/report/renderer/ReportTemplateRenderer.java

```diff
@@ -222,17 +222,28 @@ public String getExpressionSuffix(ReportDesign design) {
 	 * @return the value for the report template replacement given the initial value
 	 */
 	public Object getReplacementValue(Object initialValue) {
+
 		Object replacementValue = "";
-		if (initialValue != null) { 
-			if (initialValue instanceof Cohort) {
-				replacementValue = new Integer(((Cohort) initialValue).size());
-			} 
-			else if (initialValue instanceof IndicatorResult) {
-				replacementValue = new Double(((IndicatorResult) initialValue).getValue().doubleValue());
-			}
-			else {
-				replacementValue = initialValue;
+		if (initialValue != null) {
+				if (initialValue instanceof Cohort) {
+					replacementValue = new Integer(((Cohort) initialValue).size());
+				}
+				else if (initialValue instanceof IndicatorResult) {

```

Comment:
There doesn't seem to be much point in the Integer and Double checking as written, as in all of the cases within the IndicatorResult block, you set the replacementValue = ir.getValue().  The cast is not adding additional value unless you use it for something, which we are not.  So I think you just return ir.getValue() in all cases where it is an IndicatorResult.

Reply:
@mseaton this is done, I have removed the casts for both integers and Double

Reply:
@solemabrothers what i meant is that the if/else isn't accomplishing anything below.  Why not just say:

```
else if (initialValue instanceof IndicatorResult) {
    IndicatorResult ir = (IndicatorResult) initialValue;
    replacementValue = ir.getValue();
}
```

Reply:
@mseaton I am testing this now